crates/biome_deserialize/src/diagnostics.rs

```diff
@@ -5,38 +5,62 @@ use biome_diagnostics::{
     Advices, Diagnostic, DiagnosticTags, LogCategory, MessageAndDescription, Severity, Visit,
 };
 use biome_rowan::{SyntaxError, TextRange};
-use bitflags::bitflags;
+use enumflags2::{bitflags, make_bitflags, BitFlags};
 use serde::{Deserialize, Serialize};
 
-bitflags! {
-    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
-    pub struct VisitableType: u8 {
-        const NULL = 1 << 0;
-        const BOOL = 1 << 1;
-        const NUMBER = 1 << 2;
-        const STR = 1 << 3;
-        const ARRAY = 1 << 4;
-        const MAP = 1 << 5;
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[bitflags]
+#[repr(u8)]
+pub(crate) enum Types {
```

Comment:
Usually we use the singular form for enums.
```suggestion
pub(crate) enum Type {
```